emissions/cpp_util/calc_gross_emissions_generic.cpp

```diff
@@ -470,177 +459,201 @@ for(x=0; x<xsize; x++)
 			float Gef_CH4 = q[2];       // Emissions factor for CH4
 			float Gef_N2O = q[3];       // Emissions factor for N2O
 			float peatburn_CO2_only = q[4];      // Emissions from burning peat, CO2 emissions only
-			float peatburn_non_CO2 = q[5];       // Emissions from burning peat, non-CO2 emissions only
+			float peatburn_CH4_only = q[5];       // Emissions from burning peat, CH4 emissions only (there are no N2O emissions from burning peat)
     		float peat_drain_total_CO2_only = q[6];      // Emissions from draining peat, CO2 emissions only
-    		float peat_drain_total_non_CO2 = q[7];      // Emissions from draining peat, non-CO2 emissions only
+    		float peat_drain_total_CH4_only = q[7];      // Emissions from draining peat, CH4 emissions only
+    		float peat_drain_total_N2O_only = q[8];      // Emissions from draining peat, N2O emissions only
 
             // Define and calculate several values used later
 			float non_soil_c;
 			non_soil_c = agc_data[x] + bgc_data[x] + dead_data[x] + litter_data[x];
 
+			float non_soil_bgc_c;
+			non_soil_bgc_c = agc_data[x] + dead_data[x] + litter_data[x]; //used for CH4 and N2O fire emissions for forestry, wildfire, other, and no driver
```

Comment:
General note: This is used to calculate the C density for non-CO2 emissions from fires in non-land-use change drivers.